light_node/src/main.rs

```diff
@@ -516,85 +566,81 @@ fn main() {
 
     // create common RocksDB block cache to be shared among column families
     // IMPORTANT: Cache object must live at least as long as DB (returned by open_kv)
-    let cache = Cache::new_lru_cache(128 * 1024 * 1024).unwrap(); // 128 MB
-
-    let schemas = vec![
-        block_storage::BlockPrimaryIndex::descriptor(&cache),
-        block_storage::BlockByLevelIndex::descriptor(&cache),
-        block_storage::BlockByContextHashIndex::descriptor(&cache),
-        BlockMetaStorage::descriptor(&cache),
-        OperationsStorage::descriptor(&cache),
-        OperationsMetaStorage::descriptor(&cache),
-        context_action_storage::ContextActionByBlockHashIndex::descriptor(&cache),
-        context_action_storage::ContextActionByContractIndex::descriptor(&cache),
-        context_action_storage::ContextActionByTypeIndex::descriptor(&cache),
-        ContextActionStorage::descriptor(&cache),
-        MerkleStorage::descriptor(&cache),
-        SystemStorage::descriptor(&cache),
-        Sequences::descriptor(&cache),
-        MempoolStorage::descriptor(&cache),
-        ChainMetaStorage::descriptor(&cache),
-        PredecessorStorage::descriptor(&cache),
+    let cache = [
+        Cache::new_lru_cache(LRU_CACHE_SIZE_128MB).unwrap(),
+        Cache::new_lru_cache(LRU_CACHE_SIZE_128MB).unwrap(),
+        Cache::new_lru_cache(LRU_CACHE_SIZE_128MB).unwrap(),
     ];
 
-    let rocks_db = match open_kv(&env.storage.db_path, schemas, &env.storage.db_cfg) {
-        Ok(db) => Arc::new(db),
+    let storages = match initialize_databases(&cache, &env) {
         Err(e) => shutdown_and_exit!(
-            error!(log, "Failed to create RocksDB database at '{:?}'", &env.storage.db_path; "reason" => e),
+            error!(
+                log,
+                "Failed to create initialize RocksDB databases '{:?}'", e
+            ),
             actor_system
         ),
+        Ok(db) => db,
     };
-    debug!(log, "Loaded RocksDB database");
 
-    match check_database_compatibility(rocks_db.clone(), DATABASE_VERSION, &tezos_env, &log) {
-        Ok(false) => shutdown_and_exit!(
-            crit!(log, "Database incompatibility detected"),
-            actor_system
-        ),
-        Err(e) => shutdown_and_exit!(
-            error!(log, "Failed to verify database compatibility"; "reason" => e),
-            actor_system
-        ),
-        _ => (),
+    for db in storages.iter() {
+        match check_database_compatibility(db.clone(), DATABASE_VERSION, &tezos_env, &log) {
```

Comment:
@mat13mn 
I like the fix with that split SystemStorage to every subpart, I didnt think of that before :), but make sense,
so I think, that we should split also constant DATABASE_VERSION for every storage, that it make totally sense,

Reply:
@mat13mn 
and last thing, constant LRU_CACHE_SIZE_128MB I would split also for every storate, and set it somehow,
context_actions storage does not need 128, maybe just half, or just 32MB